pkg/server/sotw/v3/ads.go

```diff
@@ -101,14 +102,24 @@ func (s *server) processADS(sw *streamWrapper, reqCh chan *discovery.DiscoveryRe
 				}
 			}
 
-			if lastResponse, ok := sw.lastDiscoveryResponses[req.GetTypeUrl()]; ok {
+			typeURL := req.GetTypeUrl()
+			streamState, ok := sw.streamState[typeURL]
+			if !ok {
+				// Supports legacy wildcard mode
+				// Wildcard will be set to true if no resource is set
+				streamState = stream.NewSubscriptionState(len(req.ResourceNames) == 0, nil)
+			}
+
+			// ToDo: track ACK through subscription state
```

Comment:
nit: The project uses `TODO(name)` throughout.

```suggestion
			// TODO(valerian-roche): track ACK through subscription state
```

By the way, isn't this todo what you are doing here at L117? If not, I don't really understand this comment.

Reply:
Cleaned up, was leftover